Mathlib/Data/Multiset/Powerset.lean

```diff
@@ -23,15 +23,21 @@ variable {α : Type*}
 
 /-! ### powerset -/
 
---Porting note: TODO: Write a more efficient version
 /-- A helper function for the powerset of a multiset. Given a list `l`, returns a list
 of sublists of `l` as multisets. -/
 def powersetAux (l : List α) : List (Multiset α) :=
-  (sublists l).map (↑)
+  l.foldr (fun a acc => acc >>= (fun x => [x, a ::ₘ x])) [0]
 #align multiset.powerset_aux Multiset.powersetAux
 
-theorem powersetAux_eq_map_coe {l : List α} : powersetAux l = (sublists l).map (↑) :=
-  rfl
+theorem powersetAux_eq_map_coe {l : List α} : powersetAux l = (sublists l).map (↑) := by
```

Comment:
This would be a good place to add a comment explaining why the obvious implementation via `sublists` is undesirable